src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/TypeNamesShouldNotMatchNamespaces.cs

```diff
@@ -158,9 +158,7 @@ private static void InitializeWellKnownSystemNamespaceTable()
             {
                 lock (s_lock)
                 {
-#pragma warning disable CA1508 // Avoid dead conditional code - https://github.com/dotnet/roslyn-analyzers/issues/3861
                     if (s_wellKnownSystemNamespaceTable == null)
```

Comment:
It looks like this is still a bug. But there is a test for it:

https://github.com/dotnet/roslyn-analyzers/blob/998b9af22f12059739e64024ccbb74782be8d6c5/src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/Maintainability/AvoidDeadConditionalCode_NullAnalysis.cs#L5501

Feels weird.